swatch-producer-azure/src/main/java/com/redhat/swatch/azure/kafka/streams/BillableUsageAggregateKey.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.azure.kafka.streams;
+
+import com.redhat.swatch.azure.openapi.model.BillableUsage;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class BillableUsageAggregateKey {
+
+  private String orgId;
+
+  private String productId;
+
+  private String metricId;
+
+  private String sla;
+
+  private String usage;
+
+  private String billingProvider;
+
+  private String billingAccountId;
```

Comment:
All these fields can and should be final.

Reply:
As discussed, Jackson needs a NoArgConstructor for deserialization so we are keeping this as is for now.